webapp-django/crashstats/api/views.py

```diff
@@ -124,11 +124,12 @@ def _clear_empty_session(request, *args, **kwargs):
         # Run the function
         ret = fun(request, *args, **kwargs)
 
-        # Clear the session if nothing was saved to it
+        # Clear the session if nothing was saved to it and it was generated using a
+        # token login
         session_keys = [
             key for key in request.session.keys() if not key.startswith("_auth_user")
         ]
-        if not session_keys:
+        if not session_keys and getattr(request.user, "token_login", False):
```

Comment:
We only want to wipe the session if we know it was created because the "user" logged in using an auth token.